### PR TITLE
Add Ristretto255 support

### PR DESCRIPTION
**Pull request by Traderjoe95**
libsodium is supporting the Ristretto255 prime order group. This PR implements support in LazySodium.

### CODE REVIEW

**Comment by gurpreet-** [comment]
Libsodium has Ristretto support now? 🙂 

Reviewing now.

File: src/main/java/com/goterl/lazysodium/interfaces/Ristretto255.java [hunk 1]
```diff
@@ -0,0 +1,1206 @@
...
+                throw new IllegalArgumentException("A non-reduced Ristretto255 scalar must be "
+                                                       + RISTRETTO255_NON_REDUCED_SCALAR_BYTES
+                                                       + " bytes in size");
+            }
+        }
+    }
+
+    final class RistrettoPoint {
+
+        private final LazySodium ls;
```
Comment (gurpreet-) [inline src/main/java/com/goterl/lazysodium/interfaces/Ristretto255.java:1039]: Hm a lot of the code relies on hexadecimal strings. Perhaps we can inside `LazySodium.java` have a getter that exposes the `messageEncoder` like so:

```java
public MessageEncoder getMessageEncoder() {
    return messageEncoder;
}
```

and then you can use the message encoder to encode and decode to your liking everywhere 👍

**Review by gurpreet- — COMMENTED** [review]
Thank you for your hard work in creating this perfect PR!

I have some comments. Let's work on them together

**Comment by Traderjoe95** [comment]
Hey, sorry for the late reply, I've been having quite some busy days lately. I will address your suggested changes this week as soon as I have time 😄

**Comment by michielgkalkman** [comment]
Hi, any chance this will me merged and deployed any time soon?

**Comment by anhepol** [comment]
As @michielgkalkman says. @Traderjoe95 It would be great if this pull request could be completed. :)

**Comment by Traderjoe95** [comment]
Hey all, sorry for the massive delay. I've been completely swamped at work lately, so I couldn't really continue working on this. But I guess I will be able to finish all those changes today. Let's see.

File: src/main/java/com/goterl/lazysodium/interfaces/Ristretto255.java [hunk 2]
```diff
@@ -0,0 +1,1206 @@
...
+        // Convert big-endian to little-endian
+        byte[] temp = new byte[expectedCount];
+
+        for (int i = 0; i < bytes.length; ++i) {
+            temp[i] = bytes[bytes.length - i - 1];
+        }
+
+        bytes = temp;
+
+        return bytes;
```
[the remark this answers is not in the document]
Comment (Traderjoe95) [reply src/main/java/com/goterl/lazysodium/interfaces/Ristretto255.java]: Changed, also changed the naming for clarity

File: src/main/java/com/goterl/lazysodium/interfaces/Ristretto255.java [hunk 3]
```diff
@@ -0,0 +1,1206 @@
...
+         * @param y      the second scalar, must be {@link Ristretto255#RISTRETTO255_SCALAR_BYTES}
+         *               bytes
+         */
+        void cryptoCoreRistretto255ScalarMul(byte[] result, byte[] x, byte[] y);
+    }
+
+    interface Lazy {
+
+        /**
+         * Returns whether the passed hexadecimal string represents a valid Ristretto255 point.
```
[the remark this answers is not in the document]
Comment (Traderjoe95) [reply src/main/java/com/goterl/lazysodium/interfaces/Ristretto255.java]: Applied Message Encoder wherever possible

File: src/main/java/com/goterl/lazysodium/interfaces/Ristretto255.java (hunk 1 above)
Comment (Traderjoe95) [reply src/main/java/com/goterl/lazysodium/interfaces/Ristretto255.java:811]: * Added a new `encode` method to return the point encoded with the `LazySodium`'s `MessageEncoder`. 
* `toString` now returns the result of `encode`
* The constructor taking a string now uses the `MessageEncoder`
* `toHex` and `fromHex` methods remain as a common use case
* Added a new `fromString` method using the `MessageEncoder`
* Changed some naming
* Added the `hashToPoint` convenience method to provide a single interface for the common task of hashing some input and then calling `cryptoCoreRistretto255FromHash`

**Comment by Traderjoe95** [comment]
@gurpreet- I addressed all your comments, I hope we are clear to merge now. 😃 

Cheers!

File: src/main/java/com/goterl/lazysodium/interfaces/Ristretto255.java (hunk 1 above)
Comment (Traderjoe95) [reply src/main/java/com/goterl/lazysodium/interfaces/Ristretto255.java:811]: * Added methods `encodeToString` and `decodeFromString` using the `MessageEncoder` to `LazySodium`

**Review by gurpreet- — APPROVED** [review]

**Comment by gurpreet-** [comment]
Looks great! Merged!

**Comment by michielgkalkman** [comment]
Thanks!!

**Pull request merged**